scipy/optimize/_lsq/least_squares.py

```diff
@@ -868,8 +868,11 @@ def jac_wrapped(x, f):
                 tr_solver = 'lsmr'
 
     if method == 'lm':
-        result = call_minpack(fun_wrapped, x0, jac_wrapped, ftol, xtol, gtol,
-                              max_nfev, x_scale, diff_step)
+        if abs_step is None:
+            result = call_minpack(fun_wrapped, x0, jac_wrapped, ftol, xtol, gtol,
+                                  max_nfev, x_scale, rel_step)
+        else:
+            raise ValueError("'abs_step' can't be used when method='lm'")
```

Comment:
Better move this exception above, around line 726 would be good. Another minor thing: end the sentence with a period (for consistency with other error messages).